R/ensure-lib.R

```diff
@@ -86,6 +90,31 @@ ensure_lib <- function(lib = NULL, test_packages = unlist(strsplit(packageDescri
   lib_dir_path
 }
 
+install_arrow_github <- function(repo = "apache/arrow", ref = "HEAD", ...) {
+  # TODO: check if cmake is found? Say to use brew to install that?
+  # Do we want or need to also allow this to install any github package. For
+  # most it would only need to change the repo arg + expose the subdir argument
+  with_envvar(
+    list(
+      # always use the tools/linuxlibs.R script for installing arrow lib
+      USE_TOOLS_LIBS_SCRIPT = "true",
+      # we want everything to be available
+      LIBARROW_MINIMAL = "false",
+      # Don't use pkgconfig because we don't want to find sources that are already installed
+      ARROW_USE_PKG_CONFIG = "false",
```

Comment:
I think you'll want the new "use script" flag to set this value in `configure` too, so you shouldn't need to set it here.